### PR TITLE
[Storage] Support container softdelete

### PR DESCRIPTION
**Pull request by blueww**
## Description
design pass review in: Azure/azure-powershell-cmdlet-review-pr#726

Add a static analysis exception for -RetentionDays. This is already discussed in design review and the name agreed.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageBlobTests.ps1 [hunk 1]
```diff
@@ -757,3 +757,65 @@ function Test-StorageBlobChangeFeed
...
+		Assert-AreEqual $true $cons[0].Deleted
+
+
+		# Disable Blob Delete Retention Policy
+		$policy = Disable-AzStorageContainerDeleteRetentionPolicy -ResourceGroupName $rgname -StorageAccountName $stoname  -PassThru
+		Assert-AreEqual $false $policy.Enabled
+		$property = Get-AzStorageBlobServiceProperty -ResourceGroupName $rgname -StorageAccountName $stoname
+		Assert-AreEqual $false $property.ContainerDeleteRetentionPolicy.Enabled
+
+        Remove-AzStorageAccount -Force -ResourceGroupName $rgname -Name $stoname;
```
Comment (BethanyZhou) [inline src/Storage/Storage.Management.Test/ScenarioTests/StorageBlobTests.ps1:812]: HI @blueww , please correct indents.

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1 [hunk 2]
```diff
@@ -301,6 +301,30 @@ function Test-Blob
 		Assert-AreEqual $scopename2 $blob.BlobProperties.EncryptionScope
 		Remove-AzStorageContainer -Name $containerName2 -Force -Context $storageContext
 
+		# container softdelete test
+		## Enabled container softdelete,then create and delete a container
+		Enable-AzStorageContainerDeleteRetentionPolicy -ResourceGroupName $ResourceGroupName -Name $StorageAccountName -RetentionDays 3
+		$containerNamesoftdelete = "softdeletecontainer"
+		New-AzStorageContainer -Name $containerNamesoftdelete -Context $storageContext
+		Remove-AzStorageContainer -Name $containerNamesoftdelete -Context $storageContext -Force
+		## Get container without -IncludeDeleted, won't list out deleted containers
```
Comment (BethanyZhou) [inline src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1:310]: Please correct indents.

**Review by BethanyZhou — COMMENTED** [review]
Hi @blueww , please submit a cmdlet design for new added cmdlets.
For `Enable-AzStorageContainerDeleteRetentionPolicy` and `Disable-AzStorageContainerDeleteRetentionPolicy`, 
I suggest we combine them as one cmdlet, for instance `Update-AzStorageContainer [DeleteRetentionPolicy <switch>]`.

**Comment by blueww** [comment]
@BethanyZhou 

For Enable-AzStorageContainerDeleteRetentionPolicy and Disable-AzStorageContainerDeleteRetentionPolicy, they are alredy in https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/726. 
I would suggest not combine them, since:
1. We use enable/disable-* cmdlets for many other blob service properties like blob softdelete, and we would like to keep aligned. 
2. We already release this feature in preview module, and would like to keep same behavior with preview module to avoid user breaking

**Comment by BethanyZhou** [comment]
Make sense.

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1 (hunk 2 above)
Comment (blueww) [reply src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1:310]: fixed

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageBlobTests.ps1 (hunk 1 above)
Comment (blueww) [reply src/Storage/Storage.Management.Test/ScenarioTests/StorageBlobTests.ps1]: fixed

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**